content/docs/5.1/concepts/explore-tikv-features/fault-tolerance.md

```diff
@@ -1,10 +1,220 @@
 ---
-title: Fault tolerance
-description: TODO
+title: Fault Tolerance and Recovery
+description: Learn how TiKV recovers from failures
 menu:
     "5.1":
         parent: Features
         weight: 2
 ---
 
-TODO
+This page walks you through a simple demonstration of how TiKV recovers from failures and continues to provide services during failures.
+
+1. Starting a 6-node local cluster.
+2. Run a sample workload via [go-ycsb](https://github.com/pingcap/go-ycsb), terminate a node to simulate a failure, and see how the cluster continues uninterrupted.
+3. Leave that node offline for long enough to watch the cluster repair itself by re-replicating missing data to other nodes.
+4. Prepare the cluster for 2 simultaneous node failures by increasing to 5-way replication, then take two nodes offline at the same time, and again see how the cluster continues uninterrupted.
```

Comment:
```suggestion
4. Increase to 5 replications, then simultaneously take two nodes offline, and again see how the cluster recovers from failure without interrupting online service .
```

Reply:
Ditto.

Reply:
This made the sentence more clear by reducing unnecessary long words and put the action you want to do to the very beginning.

Reply:
Accepting the former suggestion. 
```suggestion
4. Increase to 5 replications, then simultaneously take two nodes offline, and again see how the cluster continues uninterrupted.
```